open-machine-learning-jupyter-book/ml-fundamentals/classification/more-classifiers.md

```diff
@@ -14,3 +14,221 @@ kernelspec:
+width: 90%
+---
+Image by [Jen Looper](https://twitter.com/jenlooper)
+```
+
+```{note}
+A section of Microsoft's Algorithm Cheat Sheet, detailing multiclass classification options.
+```
+
+```{seealso}
+Download this cheat sheet, print it out, and hang it on your wall!
+```
+
+### Reasoning
+
+Let's see if we can reason our way through different approaches given the constraints we have:
+
+- **Neural networks are too heavy**. Given our clean, but minimal dataset, and the fact that we are running training locally via notebooks, neural networks are too heavyweight for this task.
+- **No two-class classifier**. We do not use a two-class classifier, so that rules out one-vs-all. 
+- **Decision tree or logistic regression could work**. A decision tree might work, or logistic regression for multiclass data. 
+- **Multiclass Boosted Decision Trees solve a different problem**. The multiclass boosted decision tree is most suitable for nonparametric tasks, e.g. tasks designed to build rankings, so it is not useful for us.
+
+### Using Scikit-learn 
+
+We will be using Scikit-learn to analyze our data. However, there are many ways to use logistic regression in Scikit-learn. Take a look at the [parameters to pass](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html?highlight=logistic%20regressio#sklearn.linear_model.LogisticRegression).  
+
+Essentially there are two important parameters - `multi_class` and `solver` - that we need to specify, when we ask Scikit-learn to perform a logistic regression. The `multi_class` value applies a certain behavior. The value of the solver is what algorithm to use. Not all solvers can be paired with all `multi_class` values.
+
+According to the docs, in the multiclass case, the training algorithm:
+
+- **Uses the one-vs-rest (OvR) scheme**, if the `multi_class` option is set to `ovr`
+- **Uses the cross-entropy loss**, if the `multi_class` option is set to `multinomial`. (Currently the `multinomial` option is supported only by the ‘lbfgs’, ‘sag’, ‘saga’ and ‘newton-cg’ solvers.)"
+
+```{seealso}
+The 'scheme' here can either be 'ovr' (one-vs-rest) or 'multinomial'. Since logistic regression is really designed to support binary classification, these schemes allow it to better handle multiclass classification tasks. [source](https://machinelearningmastery.com/one-vs-rest-and-one-vs-one-for-multi-class-classification/)
+
+The 'solver' is defined as "the algorithm to use in the optimization problem". [source](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html?highlight=logistic%20regressio#sklearn.linear_model.LogisticRegression).
+```
+
+Scikit-learn offers this table to explain how solvers handle different challenges presented by different kinds of data structures:
+
+```{figure} ../../../images/ml-fundamentals/ml-classification/solvers.png
+---
+name: 'solvers'
+width: 90%
+---
+Image by [Jen Looper](https://twitter.com/jenlooper)
+```
```

Comment:
Let's use a real HTML or markdown table. How to generate a table: 
- https://www.tablesgenerator.com/html_tables
- https://www.tablesgenerator.com/markdown_tables

Reply:
Here, the table is a picture.